src/test/shared/utilities/workspaceUtils.test.ts

```diff
@@ -0,0 +1,82 @@
+            name: 'tempFolder',
+            index: 0,
+            uri: vscode.Uri.parse(workspaceDir),
+        })
+    })
+    const sourceCodeUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'src', 'Program.cs'))
+    const projectInSameFolderUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'src', 'App.csproj'))
+    const projectInParentFolderUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'App.csproj'))
+    const projectInParentParentFolderUri = vscode.Uri.file(path.join(workspaceDir, 'App.csproj'))
+    const projectInParentButOutOfWorkspace = vscode.Uri.file(path.join('code', 'App.csproj'))
+    const projectOutOfParentChainUri = vscode.Uri.file(path.join('code', 'someotherproject', 'App.csproj'))
+
+    const testScenarios = [
+        {
+            scenario: 'locates project in same folder',
+            findFilesResult: [projectInSameFolderUri],
+            expectedResult: projectInSameFolderUri,
+        },
+        {
+            scenario: 'locates project in parent folder',
+            findFilesResult: [projectInParentFolderUri],
+            expectedResult: projectInParentFolderUri,
+        },
+        {
+            scenario: 'locates project two parent folders up',
+            findFilesResult: [projectInParentParentFolderUri],
+            expectedResult: projectInParentParentFolderUri,
+        },
+        {
+            scenario: 'selects project in same folder over parent folder',
+            findFilesResult: [projectInSameFolderUri, projectInParentFolderUri],
+            expectedResult: projectInSameFolderUri,
+        },
+        {
+            scenario: 'always selects project in same folder over parent folder regardless of order',
+            findFilesResult: [projectInParentFolderUri, projectInSameFolderUri],
+            expectedResult: projectInSameFolderUri,
+        },
+        {
+            scenario: 'returns undefined when no project files are located',
+            findFilesResult: [],
+            expectedResult: undefined,
+        },
+        {
+            scenario: 'returns undefined when no project files are located in parent chain',
+            findFilesResult: [projectOutOfParentChainUri],
+            expectedResult: undefined,
+        },
+        {
+            scenario: 'returns undefined when the project file is in the parent chain but out of the workspace folder',
+            findFilesResult: [projectInParentButOutOfWorkspace],
+            expectedResult: undefined,
+        },
+    ]
+
+    testScenarios.forEach(test => {
+        it(test.scenario, async () => {
+            const projectFile = await findParentProjectFile(
+                sourceCodeUri,
+                path.join('**', '*.csproj'),
+                async (): Promise<vscode.Uri[]> => test.findFilesResult
+            )
+            assert.strictEqual(projectFile, test.expectedResult, 'Project file was not the expected one')
```

Comment:
is "Project file was not the expected one" adding any insight not already made clear by the method name `strictEqual()` + the detailed info given by the test framework on failure?

Reply:
Probably not, I'll remove it.